snap/snapenv/snapenv_test.go

```diff
@@ -295,10 +295,16 @@ func (s *HTestSuite) TestHiddenDirEnv(c *C) {
 	testDir := c.MkDir()
 	usr.HomeDir = testDir
 
-	restore := MockUserCurrent(func() (*user.User, error) {
+	// The code is kept for comparison with the old mocking technique.
+	// The commented-out version of MockUserCurrent is preserved below in this file.
+	//restore := MockUserCurrent(func() (*user.User, error) {
+	//	return usr, nil
+	//})
+	//defer restore()
+
+	s.MockAny(&userCurrent, func() (*user.User, error) {
```

Comment:
this is possible because this test is in "package snapenv" but we prefer nowdays to follow the practice to put tests in _test packages at which point this kind of pattern is not possible because then userCurrent wouldn't be accessible. So I think the helper on BaseTest is counterproductive because it helps to write code that we don't consider good practice anymore.

Reply:
For that purpose there is the other API that can be used instead.